src/Microsoft.Data.SqlClient/tests/ManualTests/TracingTests/EventCounterTest.cs

```diff
@@ -16,91 +20,250 @@ namespace Microsoft.Data.SqlClient.ManualTesting.Tests
     /// </summary>
     public class EventCounterTest
     {
+        public EventCounterTest()
+        {
+            ClearConnectionPools();
+        }
+
+        // [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        // public void EventCounterTestAll()
+        // {
+        //     var stringBuilder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString)
+        //     {
+        //         Pooling = true, MaxPoolSize = 20
+        //     };
+        //
+        //     OpenConnections(stringBuilder.ConnectionString);
+        //     stringBuilder.Pooling = false;
+        //     OpenConnections(stringBuilder.ConnectionString);
+        //
+        //     Thread.Sleep(3000);
+        //     _fixture.Listener.WaitForStatsUpdated();
+        //
+        //     //there are 16 counters total - all of them must have been collected
+        //     Assert.Equal(16, _fixture.Listener.EventCounters.Count);
+        // }
+
+        // private void OpenConnections(string cnnString)
+        // {
+        //     List<Task> tasks = new List<Task>();
+        //
+        //     Enumerable.Range(1, 100).ToList().ForEach(i =>
+        //     {
+        //         SqlConnection cnn = new SqlConnection(cnnString);
+        //         cnn.Open();
+        //         int x = i;
+        //         tasks.Add(Task.Run(() =>
+        //         {
+        //             Thread.Sleep(x);
+        //             cnn.Close();
+        //         }));
+        //     });
+        //     Task.WhenAll(tasks).Wait();
+        // }
+
         [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
-        public void EventCounterTestAll()
+        public void EventCounter_HardConnectionsCounters_Functional()
         {
-            var stringBuilder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString)
-            {
-                Pooling = true,
-                MaxPoolSize = 20
-            };
+            //create a non-pooled connection
+            var stringBuilder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) {Pooling = false};
 
-            using (var TraceListener = new TraceEventCounterListener())
-            {
-                OpenConnections(stringBuilder.ConnectionString);
-                stringBuilder.Pooling = false;
-                OpenConnections(stringBuilder.ConnectionString);
+            using var conn = new SqlConnection(stringBuilder.ToString());
```

Comment:
Can we use the using block for SqlConnection here?

Reply:
missing using block

Reply:
Done